keanu-project/src/main/java/io/improbable/keanu/vertices/dbltensor/probabilistic/TensorUniformVertex.java

```diff
@@ -84,15 +85,15 @@ public double logPdf(DoubleTensor value) {
     public Map<Long, DoubleTensor> dLogPdf(DoubleTensor value) {
 
         DoubleTensor dlogPdf = DoubleTensor.zeros(this.xMax.getValue().getShape());
-        dlogPdf.applyWhere(value.getGreaterThanMask(xMax.getValue()), Double.NEGATIVE_INFINITY);
-        dlogPdf.applyWhere(value.getLessThanOrEqualToMask(xMin.getValue()), Double.POSITIVE_INFINITY);
+        dlogPdf.applyWhereInPlace(value.getGreaterThanMask(xMax.getValue()), Double.NEGATIVE_INFINITY);
```

Comment:
Nice work with `applyWhereInPlace`. I can see a use case in which we'd want to have `applyWhere`. Add that in too?

Reply:
Good call. I've added a non-in-place version and changed the name to setWithMask in order to avoid confusion with the apply method you're adding in your Gamma PR.